models/marts/dim_contract_code_hist.sql

```diff
@@ -0,0 +1,242 @@
+4. Supports multi-day catch-up scenarios by processing all relevant records based on the incremental logic.
+
+Usage:
+------
+Compile:
+    dbt compile --models dim_contract_code_hist
+
+Run:
+    dbt run --models dim_contract_code_hist --full-refresh
+    dbt run --models dim_contract_code_hist
+*/
+
+-- Load latest changes from transformed contract code data, applying incremental filter
+with
+    source_data as (
+        select
+            contract_code_hash
+            , ledger_key_hash
+            , n_instructions
+            , n_functions
+            , n_globals
+            , n_table_entries
+            , n_types
+            , n_data_segments
+            , n_elem_segments
+            , n_imports
+            , n_exports
+            , n_data_segment_bytes
+            , contract_create_ts
+            , contract_delete_ts
+            , closed_at
+            , date(closed_at) as start_date
+            , airflow_start_ts
+            , batch_id
+            , batch_run_date
+            , row_hash
+        from {{ ref('int_transform_contract_code') }}
+        {% if is_incremental() %}
+            where closed_at > (select coalesce(max(closed_at), '2000-01-01T00:00:00+00:00') from {{ this }})
+        {% endif %}
+    )
+
+    -- Get existing history records for affected contracts, or create empty shell for initial load
+    , target_data as (
+        {% if is_incremental() %}
+            select *
+            from {{ this }}
+            where ledger_key_hash in (select distinct ledger_key_hash from source_data)
+        {% else %}
+        select
+              CAST(NULL AS STRING) AS ledger_key_hash
+            , CAST(NULL AS DATE) AS start_date
+            , CAST(NULL AS STRING) AS contract_code_hash
+            , CAST(NULL AS INT64) AS n_instructions
+            , CAST(NULL AS INT64) AS n_functions
+            , CAST(NULL AS INT64) AS n_globals
+            , CAST(NULL AS INT64) AS n_table_entries
+            , CAST(NULL AS INT64) AS n_types
+            , CAST(NULL AS INT64) AS n_data_segments
+            , CAST(NULL AS INT64) AS n_elem_segments
+            , CAST(NULL AS INT64) AS n_imports
+            , CAST(NULL AS INT64) AS n_exports
+            , CAST(NULL AS INT64) AS n_data_segment_bytes
+            , CAST(NULL AS TIMESTAMP) AS contract_create_ts
+            , CAST(NULL AS TIMESTAMP) AS contract_delete_ts
+            , CAST(NULL AS TIMESTAMP) AS closed_at
+            , CAST(NULL AS DATE) AS end_date
+            , CAST(NULL AS BOOLEAN) AS is_current
+            , CAST(NULL AS TIMESTAMP) AS airflow_start_ts
+            , CAST(NULL AS STRING) AS batch_id
+            , CAST(NULL AS DATETIME) AS batch_run_date
+            , CAST(NULL AS STRING) AS row_hash
+            , CAST(NULL AS TIMESTAMP) AS dw_load_ts
+            , CAST(NULL AS TIMESTAMP) AS dw_update_ts
+        from source_data
+        where 1 = 0
```

Comment:
Doesn't this always evaluate to false?

Reply:
- You are right that this always evaluates to false. That is intention and will always produce as zero-row result set with schema.

- It doesn't insert any Null rows and will always produce zero rows

- Primary purpose is to allow the same code downstream to work during Initial Load/ Full Refresh as well as Incremental Load. Without this, we have to write customized code to execute with is_incremental() blocks.

Reply:
Does
```
select * from {{ this }}
```
evaluate as an empty table with the correct schema on a full_refresh?

Reply:
I tried using that but during first run or whenever we use full-refresh, its dropping the table and doesn't have any reference of what {{ this }} means and is failing.